### PR TITLE
Rename the pkg to "treeple"

### PR DESCRIPTION
**Pull request by SUKI-O**
#### Reference Issues/PRs

Change the package name from `scikit-tree` to `treeple`

#### What does this implement/fix? Explain your changes.
* Replace `scikit-tree` to `treeple`
* Replace `sktree` to `treeple`
* Change the directory name `sktree` to `treeple`

#### Any other comments?
* The name of the repo itself in GH also need to be changed when ready
* After that, pypi has to be updated (I need help with this part)
* It builds locally using `spin` as expected.  
* When trying locally, I believe you have to re-initialize the submodule
```
git submodule init
git submodule update
```

### CODE REVIEW

**Comment by adam2392** [comment]
Mostly LGTM, but 

1. you'll need to fix the building of the documentation (https://app.circleci.com/pipelines/github/neurodata/scikit-tree/1741/workflows/345d4498-1b23-4ca7-8aaa-4dedf03eac94/jobs/2969?utm_campaign=vcs-integration-link&utm_medium=referral&utm_source=github-checks-link&utm_content=summary)
2. in the CHANGELOG entry, you should add a summary specifying the entire package was renamed. https://github.com/neurodata/scikit-tree/blob/98b67fcbaf028ee8a9bd92407089ab4a34008e9b/doc/whats_new/v0.9.rst?plain=1#L11
3. In the README, it would also be good to specify that in v0.9 and onwards, the package was renamed to treeple. <v0.8 is still installable via `scikit-tree` on pip.

**Comment by SUKI-O** [comment]
@adam2392 I think the error is bc the repo name hasn't changed yet.. I think it should be fine once rename the repo itself? LMK what you think.

File: README.md [hunk 1]
```diff
@@ -8,12 +8,14 @@
...
 
 treeple
-===========
+=======
 
 treeple is a scikit-learn compatible API for building state-of-the-art decision trees. These include unsupervised trees, oblique trees, uncertainty trees, quantile trees and causal trees.
 
 Tree-models have withstood the test of time, and are consistently used for modern-day data science and machine learning applications. They especially perform well when there are limited samples for a problem and are flexible learners that can be applied to a wide variety of different settings, such as tabular, images, time-series, genomics, EEG data and more.
 
+Note that this package was originally named ``scikit-tree`` but was renamed to ``treeple`` after version 0.9.0. version <0.9.0 is still available at <https://pypi.org/project/scikit-tree/>.
```
Comment (adam2392) [inline README.md]: This should state v0.8.0 as by v0.9.0 we will rename I suppose?

File: .circleci/config.yml [hunk 2]
```diff
@@ -55,13 +55,13 @@ commands:
             echo ${CI_PULL_REQUEST//*pull\//} | tee merge.txt
             if [[ $(cat merge.txt) != "" ]]; then
               echo "Merging $(cat merge.txt)";
-              git remote add upstream https://github.com/neurodata/scikit-tree.git;
+              git remote add upstream https://github.com/neurodata/treeple.git;
```
Comment (adam2392) [inline .circleci/config.yml]:
```suggestion
              git remote add upstream https://github.com/neurodata/scikit-tree.git;
```

**Comment by adam2392** [comment]
@SUKI-O I think we can make that final change once we confirm the CI functions as expected. Seems like there are still some leftover warnings. You can try resolving locally too

File: doc/whats_new/v0.9.rst [hunk 3]
```diff
@@ -13,7 +13,8 @@ Version 0.9
 Changelog
 ---------
 
-- 
+- |MNT| Rename the package to ``treeple``. By `SUKI-O`_ (:pr:`#292`)
```
Comment (adam2392) [inline doc/whats_new/v0.9.rst]:
```suggestion
- |API| Rename the package to ``treeple``. By `SUKI-O`_ (:pr:`#292`)
```

This is a major API change.

File: doc/whats_new/v0.9.rst
Comment (adam2392) [inline doc/whats_new/v0.9.rst:1]: Under `Version 0.9` title, you can add a sentence saying that the package was renamed to `treeple`, and therefore someone can replace

`import sktree` with `import treeple as tp`

and

`from sktree import ...` with `from treeple import …`
`from sktree… import ...` with `from treeple... import ...`

**Review by adam2392 — CHANGES_REQUESTED** [review]
You need to rename the `sklearn_vs_sktree` folder under `examples/` to `sklearn_vs_treeple` to fix the CI error

**Review by adam2392 — APPROVED** [review]
LGTM. Thanks @SUKI-O!

**Pull request merged**